network-api/networkapi/wagtailpages/utils.py

```diff
@@ -6,14 +6,14 @@ def set_main_site_nav_information(page, context, homepage_class_name):
     # Find the homepage, and then record all pages that should end up as nav items. Note
     # that subclasses can bypass this, because the MozfestHomepage doesn't need any of
     # this work to be done.
-    homepage = list(filter(
-        lambda x: x.specific.__class__.__name__ is homepage_class_name,
-        page.get_ancestors()
-    ))[0]
 
-    context['menu_root'] = homepage
-    context['menu_items'] = homepage.get_children().live().in_menu()
+    root = list(filter(
+        lambda x: x.specific.__class__.__name__ == homepage_class_name,
+        page.get_ancestors(True)  # incude ourselves!
```

Comment:
Will this affect other page classes which use this method? e.g., `IndexPage`, `ModularPage`, `PrimaryPage`?

Reply:
we have a dedicated `Homepage` class for the foundation site as well, so this should not interfere.